src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,79 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.exception.UserNotAllowed;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+
+    @BeforeEach
+    void setup() {
+        user = new User();
+        user.setLogin("volodymyr.nyshta@gmail.com");
+        user.setAge(21);
+        user.setPassword("123Test");
+    }
+
+    @Test
+    void isCorrectType() {
+        assertTrue(user instanceof User);
+    }
+
+    @Test
+    void userWithTheSameLoginAlreadyExists_NotOk() {
+        User user2 = new User();
+        user2.setAge(18);
+        user2.setLogin("volodymyr.nyshta@gmail.com");
+        user2.setPassword("1234567");
+        registrationService.register(user);
+        assertThrows(UserNotAllowed.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    void userNullNotOk() {
+        User actual = registrationService.register(null);
+        assertNull(actual);
+    }
+
+    @Test
+    void loginIsNotOk() {
```

Comment:
The `loginIsNotOk` test should specify what makes the login not OK. For example, if it's the length of the login that's the issue, the test name should reflect that. Moreover, there's no check on what makes 'test' an invalid login according to the requirements.